src/Operation/FindOneAndUpdate.php

```diff
@@ -108,20 +109,16 @@ public function __construct($databaseName, $collectionName, $filter, $update, ar
             throw new InvalidArgumentException('Expected an update document with operator as first key or a pipeline');
         }
 
-        $options += [
-            'returnDocument' => self::RETURN_DOCUMENT_BEFORE,
-            'upsert' => false,
-        ];
-
         if (isset($options['projection']) && ! is_array($options['projection']) && ! is_object($options['projection'])) {
             throw InvalidArgumentException::invalidType('"projection" option', $options['projection'], 'array or object');
         }
 
-        if (! is_integer($options['returnDocument'])) {
+        if (array_key_exists('returnDocument', $options) && ! is_integer($options['returnDocument'])) {
```

Comment:
The main idea of this check is to keep the same behavior of constructor verification stage. Previously there was an assumption that the array definitely contains field 'returnDocument', so the old check covered all values include NULL value.